bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/ExpressionPart.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.common;
+
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Date;
+
+/**
+ * <code>ExpressionPart</code> are the building blocks that make up an {@link Expression}. Each part can be set, then
+ * parsed and be applied to the set of candidate dates that finally constitute the list of dates the expression will
+ * adhere to
+ *
+ * @author Karel Goderis
+ *
+ */
+interface ExpressionPart extends Comparable<ExpressionPart> {
+
+    /**
+     * Set the input string for the part of the expression
+     *
+     * @param s
+     */
+    public void set(String s);
```

Comment:
Here it could be removed I think. Unless, for another kind of Expression, specific setters and getters are defined to change part of the definition. In that case, the existing ExpressionPart should be re-used. Since the ExpressionParts are created and added to an ArrayList, I prefer them to be changed instead of looking for the correct ExpressionPart in the list, remove it, create a new one and add that one instead...